internal/codecs/nil.go

```diff
@@ -0,0 +1,31 @@
+// This source file is part of the EdgeDB open source project.
+//
+// Copyright 2020-present EdgeDB Inc. and the EdgeDB authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package codecs
+
+import (
+	"unsafe"
+
+	"github.com/edgedb/edgedb-go/internal/buff"
+	"github.com/edgedb/edgedb-go/internal/descriptor"
+	types "github.com/edgedb/edgedb-go/internal/edgedbtypes"
+)
+
+type nilDecoder struct{}
+
+func (c nilDecoder) DescriptorID() types.UUID { return descriptor.NilID }
+
+func (c nilDecoder) Decode(r *buff.Reader, out unsafe.Pointer) {}
```

Comment:
Hm, this seems to be something that doesn't decode literal `NULL`/empty sets into `nil`, it's just a codec that does nothing, is that right?

Reply:
Correct it decodes empty blocks corresponding to descriptor id 0. I'm open to other names. I didn't think too hard about that one.

Reply:
I'd call it "NoOpDecoder"

Reply:
perfect! I came to the same conclusion independently

changed